File: 2.1/LPC/s2/c.py
n = int(input())

query = []
dir = 1
while(n):
    n -= 1
    what = list(input().split())
    if what[0] == 'back':
        if query:
            if dir:
                print(query.pop())
            else:
                print(query.pop(0))
        else:
            print('No job for Ada?')
    elif what[0] == 'front':
        if query:
            if dir:
                print(query.pop(0))
            else:
                print(query.pop())
        else:
            print('No job for Ada?')
    elif what[0] == 'reverse':
        if dir:
            dir = 0
        else:
            dir = 1
    elif what[0] == 'push_back':
        if dir:
            query += [int(what[1])]
        else:
            query = [int(what[1])] + query
    elif what[0] == 'toFront':
        if dir:
            query = [int(what[1])] + query
        else:
            query += [int(what[1])]
        

File: 2.1/LPC/s2/c.cpp
#include<bits/stdc++.h> 

using namespace std;

int main() {
    int num;
    scanf("%d", &num);
    deque <int> queue;
    while(num--) {
        int dir = 0;
        char str[100];
        scanf("%s", str);

        if(!strcmp(str, "back")) {
            if(queue.empty()) {
                printf("No job for Ada?\n");
            }
            else {
                if(!dir) {
                    cout << queue.front() << endl;
                    queue.pop_front();
                }
                else {
                    cout << queue.back() << endl;
                    queue.pop_back();
                }
            }
        }
        else if(!strcmp(str, "front")) {
            if(queue.empty()) {
                printf("No job for Ada?\n");
            }
            else {
                if(!dir) {
                    cout << queue.back() << endl;
                    queue.pop_back();
                }
                else {
                    cout << queue.front() << endl;
                    queue.pop_front();
                }
            }
        }
        else if(!strcmp(str, "reverse")){
            if(dir){
                dir = 0;
            } else {
                dir = 1;
            }
        }
        else if(!strcmp(str, "push_back")) {
            int aux;
            cin >> aux;
            if(dir){
                queue.push_front(aux);
            }
            else{
                queue.push_back(aux);
            } 
        }
        else if(!strcmp(str, "toFront")) {
            int aux;
            cin >> aux;
            if(dir){
                queue.push_back(aux);
            }
            else{
                queue.push_front(aux);
            } 
        }
    }
}